packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -205,44 +354,52 @@ def reduce_logging():
 
 @pytest.mark.skipif(not lb_enabled(), reason='Load Balancer disabled')
 def test_vip(dcos_api_session, reduce_logging):
-    '''Test VIPs between the following source and destination configurations:
-        * containers: DOCKER, UCR and NONE
-        * networks: USER, BRIDGE (docker only), HOST
-        * agents: source and destnations on same agent or different agents
-        * vips: named and unnamed vip
-
+    '''Test every permutation of VIP
     '''
-    addrs = [['1.1.1.{}:{}', '1.1.1.{}:{}'],
-             ['/namedvip{}:{}', 'namedvip{}.marathon.l4lb.thisdcos.directory:{}']]
     # tests
     # UCR doesn't support BRIDGE mode
-    permutations = [[c, vi, va, sh, vn, pn]
+    permutations = [[c, ss, sh, vn, pn]
                     for c in ['NONE', 'UCR', 'DOCKER']
-                    for [vi, va] in addrs
+                    for ss in [True, False]
                     for sh in [True, False]
                     for vn in ['USER', 'BRIDGE', 'HOST']
                     for pn in ['USER', 'BRIDGE', 'HOST']]
-    tests = [VipTest(i, c, vi, va, sh, vn, pn) for i, [c, vi, va, sh, vn, pn] in enumerate(permutations)]
+    tests = [VipTest(i + 1, c, ss, sh, vn, pn) for i, [c, ss, sh, vn, pn] in enumerate(permutations)]
     executor = concurrent.futures.ThreadPoolExecutor(max_workers=maxthreads)
     # deque is thread safe
-    failed_tests = deque(tests)
+    # tests that didn't finish running
+    failed_tests = deque()
+    # test that ran successfully
     passed_tests = deque()
+    # tests that were skipped because cluster configuration cannot support them
     skipped_tests = deque()
     # skip certain tests
-    for r in tests:
-        if r.container == 'UCR' or r.container == 'NONE':
-            if r.vipnet == 'BRIDGE' or r.proxynet == 'BRIDGE':
-                r.notes = "bridge networks are not supported by mesos runtime"
-                failed_tests.remove(r)
-                skipped_tests.append(r)
+    for vip_test in tests:
+        if vip_test.container == 'UCR' or vip_test.container == 'NONE':
+            if vip_test.vipnet == 'BRIDGE' or vip_test.proxynet == 'BRIDGE':
+                vip_test.notes = 'bridge networks are not supported by mesos runtime'
+                skipped_tests.append(vip_test)
                 continue
-        if not r.samehost and len(dcos_api_session.all_slaves) == 1:
-            r.notes = "needs more then 1 agent to run"
-            failed_tests.remove(r)
-            skipped_tests.append(r)
+        if not vip_test.samehost and len(dcos_api_session.all_slaves) == 1:
+            vip_test.notes = 'needs more then 1 agent to run'
+            skipped_tests.append(vip_test)
+            continue
+        failed_tests.append(vip_test)
+    permutations = [[p, pn, vn, ss]
+                    for p in [ProxyPod.ItsOwnPod, ProxyPod.SamePodAsVip]
+                    for pn in ['HOST', 'USER']
+                    for vn in ['HOST', 'USER']
+                    for ss in [True, False]]
```

Comment:
Do we have test where two containers are sitting in the same bridge and are trying to talk to each other? Just wondering if its even feasible to write such a test in this template?